src/python/WMCore/WebTools/RESTModel.py

```diff
@@ -185,6 +185,14 @@ def wrapper(*input_args, **input_kwargs):
             if len(args) != 0:
                 # store the method name
                 method = methodKey
+                # for PUT/POST requests data may be send in body of the request
+                # we check if this is the case (no args/kwargs) and read body
+                if  verb in ['PUT', 'POST'] and \
+                    not input_args and not input_kwargs:
+                    from WMCore.Wrappers import JsonWrapper
+                    body = cherrypy.request.body.read()
+                    if  body:
+                        input_kwargs = JsonWrapper.loads( body )
```

Comment:
Sorry I wasn't clear. What I meant is that following line checks the number of argument defined by method and compare them what is passed.

https://github.com/vkuznet/WMCore/blob/9a68173916e5ef994f0453cabbfce7077f322648/src/python/WMCore/WebTools/RESTModel.py#L241

For the parameters passed by body, might not defined in the function definition.
i.e.
https://github.com/vkuznet/WMCore/blob/9a68173916e5ef994f0453cabbfce7077f322648/src/python/WMCore/HTTPFrontEnd/RequestManager/ReqMgrRESTModel.py#L133

only take one argument but if it contains message body with other parameter (i.e. message) it will fail.